### DISCUSSION TITLE
Make clear which port is used for the Web UI

### DISCUSSION THREAD

**Discussion by markg85** [music-assistant/support#2546] [Feature requests and ideas]
Hi,

Due to the recommended installation of Music Assistant (docker image with `--network host`) you don't see which port music assistant is using for the web ui. You have to consult the logs (or the readme) to figure out that 8095 is the port for the web ui.

Can this please be made a bit more discoverable?

I couldn't find how to even change this port. I'm either completely reading over it in the documentation or it's not possible?

**If** it is configurable then having it be a default filled in value in the config.json (or is this server.json?) file would already go a long way in finding this port?

This is super well hidden btw. I searched throughout the music-assistant org here on github. I know what i'm looking at is the gui in the `frontend` repo but that is apparently not the one exposing the port. I can then find the `server` repo but it talks about and **ingress** port for 8095, what the heck is "ingress" in this context? We're still talking about the Web UI here! It's probably [here](https://github.com/music-assistant/server/blob/7af1e0cbab9a4f7a4d6875eb401f4618ebd426db/music_assistant/server/controllers/webserver.py#L46) but boy o boy this is well hidden and about as vague as it can be.

**Comment by OzGav** [comment]
From the docs

> If you are running Music Assistant in docker, you need to access the webinterface at http://youripaddress:8095/.

**Reply by markg85** [reply]
As i said:
> You have to consult the logs (or the readme) to figure out that 8095 is the port for the web ui.

The reference i made to the documentation was my inability to find how to set/change it. Before you comment on that, that's **not** the question I have here. I can find what the default should be but I cannot find that in my own local installation. Hence my critique at the port being hard to discover.

I'm looking for, and I quote myself again:
> If it is configurable then having it be a default filled in value in the config.json (or is this server.json?) file would already go a long way in finding this port?

**Reply by OzGav** [reply]
8095 for the webserver is mandatory

**Reply by markg85** [reply]
Make it discoverable on the **client** side.

Could be as simple as not requiring `--network host` in docker. You'd then have to explicitly open the ports and **see** them in a normal `docker ps` command.


Surely you understand that for normal regular use you hardly ever have to go to the music assistant website. It'd be integrated in home assistant and be controlled from there. **Sometimes** you do have to fiddle in the web ui of music assistant for whatever reason. And with those sometimes I found myself googling to figure out the web ui port. Just exposing that port in either a config file or in the docker run line would already make it much easier to find the port it's using.

**Reply by marcelveldt** [reply]
The MA webserver port is set to 8095 by default. Not changeable on HA add-on (due to ingress bound to that port) but should be changeable to another port from the settings if you opened it once. I believe we document it properly.

That we require host networking has NOTHING to do with that.

**Reply by markg85** [reply]
You're missing the point. But i'll quote myself to repeat it.

> Surely you understand that for normal regular use you hardly ever have to go to the music assistant website. It'd be integrated in home assistant and be controlled from there. Sometimes you do have to fiddle in the web ui of music assistant for whatever reason. And with those sometimes I found myself googling to figure out the web ui port. Just exposing that port in either a config file or in the docker run line would already make it much easier to find the port it's using.

**Reply by OzGav** [reply]
I'm going to give you the benefit of the doubt and just let you know that you are coming across as extremely obnoxious. Your question has been answered.

**Reply by markg85** [reply]
I'm sorry if it comes across as obnoxious but what you marked as answer is not the answer i was looking for at all.
In fact, it's not a Q&A item at all anymore, it's a feature request now.

I'm asking for a way to see the port used when it's running on your local hardware.
So to make it visible that port 8095 is uses.
Referring to the docs is exactly the problem.

If you don't want to support this or can't for whatever reason then i'd be best if you just delete this question completely.

Have you never found yourself in the situation before where you're wondering: "hmm, i started this app, but which port is it on?" Using netstat to find out which port an app is using is "great". Or google to figure out what the default is. I'm sure you'd agree that being able to see it in a `docker ps <container>` command or just in a config file is a helpful addition, is it not?

**Reply by OzGav** [reply]
> hmm, i started this app, but which port is it on?

Sure but then read the docs and the answer is there. I do not agree that having to use the command line to work it out is a helpful addition.
